src/ml/federated/JasmineGraphFederatedInstance.cpp

```diff
@@ -0,0 +1,483 @@
+
+    int count = 0;
+    JasmineGraphTrainingSchedular *schedular = new JasmineGraphTrainingSchedular();
+    map<string, map<int, int>> scheduleForAllHosts = schedular->schedulePartitionTraining(graphID);
+    JasmineGraphServer *jasmineServer = new JasmineGraphServer();
+    std::map<std::string, JasmineGraphServer::workerPartitions> graphPartitionedHosts = jasmineServer->getGraphPartitionedHosts(
+            graphID);
+    int partition_count = 0;
+    std::map<std::string, JasmineGraphServer::workerPartitions>::iterator mapIterator;
+    for (mapIterator = graphPartitionedHosts.begin(); mapIterator != graphPartitionedHosts.end(); mapIterator++) {
+        JasmineGraphServer::workerPartitions workerPartition = mapIterator->second;
+        std::vector<std::string> partitions = workerPartition.partitionID;
+        std::vector<std::string>::iterator it;
+        for (it = partitions.begin(); it < partitions.end(); it++) {
+            partition_count++;
+        }
+    }
+    std::cout << partition_count << std::endl;
+    std::thread *workerThreads = new std::thread[partition_count+1];
+
+    Utils utils;
+    trainingArgs = trainingArgs;
+
+    int fl_clients = stoi(utils.getJasmineGraphProperty("org.jasminegraph.fl_clients"));
+
+    workerThreads[0] = std::thread(initiateServer,"localhost", 7780, 7781,trainingArgs,fl_clients, "1");
+
+    count = 1;
+
+    std::map<std::string, JasmineGraphServer::workerPartitions>::iterator j;
+    for (j = graphPartitionedHosts.begin(); j != graphPartitionedHosts.end(); j++) {
+        JasmineGraphServer::workerPartitions workerPartition = j->second;
+        std::vector<std::string> partitions = workerPartition.partitionID;
+        string partitionCount = std::to_string(partitions.size());
+        std::vector<std::string>::iterator k;
+        map<int, int> scheduleOfHost = scheduleForAllHosts[j->first];
+        for (k = partitions.begin(); k != partitions.end(); k++) {
+            int iterationOfPart = scheduleOfHost[stoi(*k)];
+            workerThreads[count] = std::thread(initiateClient,"localhost", workerPartition.port, workerPartition.dataPort,trainingArgs + " " + *k, fl_clients, *k);
+            count++;
+        }
+    }
+
+    for (int threadCount = 0; threadCount < count; threadCount++) {
+        workerThreads[threadCount].join();
+        std::cout << "Thread " << threadCount << " joined" << std::endl;
+    }
+
+    std::cout << "Federated learning commands sent" <<std::endl;
+
+}
+
+
+void JasmineGraphFederatedInstance::initiateMerge(std::string graphID, std::string trainingArgs,SQLiteDBInterface sqlite) {
+
+
+    int count = 0;
+    JasmineGraphTrainingSchedular *schedular = new JasmineGraphTrainingSchedular();
+    map<string, map<int, int>> scheduleForAllHosts = schedular->schedulePartitionTraining(graphID);
+    JasmineGraphServer *jasmineServer = new JasmineGraphServer();
```

Comment:
We should not instantiate a JasmineGraph server here. We instantiate JasmineGraph server only once in the life cycle of JasmineGraph system.

Reply:
Addressed this issue in another comment

Reply:
Ok, lets keep this code as it is.